src/model/user.rs

```diff
@@ -1005,24 +1005,7 @@ impl User {
         cache_http: impl CacheHttp,
         guild_id: impl Into<GuildId>,
     ) -> Option<String> {
-        let guild_id = guild_id.into();
-
-        #[cfg(feature = "cache")]
-        {
-            if let Some(cache) = cache_http.cache() {
-                if let Some(guild) = guild_id.to_guild_cached(cache) {
-                    if let Some(member) = guild.members.get(&self.id) {
-                        return member.nick.clone();
-                    }
-                }
-            }
-        }
-
-        guild_id
-            .member(&cache_http, &self.id)
-            .await
-            .ok()
-            .and_then(|member| member.nick.as_ref().map(String::clone))
+        guild_id.into().member(cache_http, &self.id).await.ok().and_then(|member| member.nick)
```

Comment:
This is cloning the whole member instance if it was in cache, `.as_ref().map(String::clone)` should just be removed.

Reply:
Currently working on a separate PR that remove the cache clone on `member`, so then I'll have to add `member.nick.clone()` but in the meantime this is much less verbose.

Reply:
Removing the cache clone on GuildId::member is impossible, see the description of https://github.com/serenity-rs/serenity/pull/1934

Reply:
Oh, ok. That's unfortunate.